scripts/pylib/pytest-twister-harness/src/twister_harness/helpers/shell.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import logging
+import re
+import time
+
+from twister_harness.device.device_adapter import DeviceAdapter
+from twister_harness.exceptions import TwisterHarnessTimeoutException
+
+logger = logging.getLogger(__name__)
+
+
+class Shell:
+    """
+    Helper class that provides methods used to interact with shell application.
+    """
+
+    def __init__(self, device: DeviceAdapter, prompt: str = 'uart:~$', timeout: float | None = None) -> None:
+        self._device: DeviceAdapter = device
+        self.prompt: str = prompt
+        self.base_timeout: float = timeout or device.base_timeout
+
+    def wait_for_prompt(self, timeout: float | None = None) -> bool:
+        """
+        Send every 0.5 second "enter" command to the device until shell prompt
+        statement will occur (return True) or timeout will be exceeded (return
+        False).
+        """
+        timeout = timeout or self.base_timeout
+        timeout_time = time.time() + timeout
+        self._device.clear_buffer()
+        while time.time() < timeout_time:
+            self._device.write(b'\n')
+            try:
+                line = self._device.readline(timeout=0.5, print_output=False)
+            except TwisterHarnessTimeoutException:
+                # ignore read timeout and try to send enter once again
+                continue
+            if self.prompt in line:
+                logger.debug('Got prompt')
+                time.sleep(0.05)
+                self._device.clear_buffer()
+                return True
+        return False
+
+    def exec_command(self, command: str, timeout: float | None = None) -> list[str]:
```

Comment:
`print_output` can be also added, to allow user to hide logs.

Reply:
Done